apiserver/params/charms.go

```diff
@@ -3,6 +3,33 @@
 
 package params
 
+// ApplicationCharmResults contains a set of ApplicationCharmResults.
+type ApplicationCharmResults struct {
+	Results []ApplicationCharmResult `json:"results"`
+}
+
+// ApplicationCharmResult contains an ApplicationCharm or an error.
+type ApplicationCharmResult struct {
+	Result *ApplicationCharm `json:"result,omitempty"`
+	Error  *Error            `json:"error,omitempty"`
+}
+
+// ApplicationCharmInfo contains information about an
+// application's charm.
+type ApplicationCharm struct {
+	// URL holds the URL of the charm assigned to the
+	// application.
+	URL string `json:"url"`
+
+	// ForceUpgrade indicates whether or not application
+	// units should upgrade to the charm even if they
+	// are in an error state.
+	ForceUpgrade bool `json:"force-upgrade,omitempty"`
```

Comment:
Do we need this? We don't upgrade running units.

Reply:
I'm thinking we might want to use this type for the uniter, to avoid making two calls like we do now (one to get the charm URL/force flag, another to get the SHA256 sum).